src/modules/roc_sdp/parser.rl

```diff
@@ -0,0 +1,283 @@
+
+        ##### SDP FIELDS #####
+        version = digit+;
+
+        # o=<username> <sess-id> <sess-version> <nettype> <addrtype> <unicast-address>
+        origin_username = non_ws_string >start_token %{ start_p_origin_username = start_p; };
+        origin_sess_id = digit+ >start_token %{ end_p_origin_sess_id = p; };
+        origin_nettype = "IN" >start_token %{ start_p_origin_nettype = start_p; };
+
+        # Either IPv6 or IPv4
+         origin_unicast_address = non_ws_string >start_token
+            %{  end_p_origin_addr = p; } 
+            %set_origin_unicast_address;
+
+         origin_unicast_address_with_addrtype =  
+            ( "IP4" %{ cur_addrtype = address::Family_IPv4; } 
+            | "IP6" %{ cur_addrtype = address::Family_IPv6; } 
+            ) ' '  origin_unicast_address;
+
+        # action todo: sess-id should be unique for this username/host
+        origin = origin_username ' ' origin_sess_id ' '  digit+ ' ' 
+            origin_nettype ' ' origin_unicast_address_with_addrtype %set_guid;
+
+        
+        # In session-level: c=<nettype> <addrtype> <connection-address>
+        session_connection_nettype = "IN";
+        
+        # Either IPv6 or IPv4/TTL
+        session_connection_address = non_ws_string >start_token
+            %set_session_connection_data;
+
+        session_connection_with_addrtype =  
+            ( "IP4" %{ cur_addrtype = address::Family_IPv4; } 
+                ' ' session_connection_address '/' digit+
+            | "IP6" %{ cur_addrtype = address::Family_IPv6; } 
+                ' ' session_connection_address
+            );
+
+        session_connection_data = session_connection_nettype ' ' 
+            session_connection_with_addrtype;
+    
+        # Each media description starts with an "m=" field and is terminated by
+        # either the next "m=" field or by the end of the session description
+        # m=<type> <port> <proto> <fmt>
+
+        # Typically "audio", "video", "text", or "application"
+        media_type = ("audio"  %{ result.last_media_description()->set_type(sdp::MediaType_Audio); } 
+            | "video" %{ result.last_media_description()->set_type(sdp::MediaType_Video); }
+            | "text" %{ result.last_media_description()->set_type(sdp::MediaType_Text); }
+            | "application" %{ result.last_media_description()->set_type(sdp::MediaType_Application); }
+        );
+            
+        
+        # Typically "RTP/AVP".
+        media_proto = "RTP/AVP"  %{ result.last_media_description()->set_proto(sdp::MediaProto_RTP_AVP); };
+
+        media_port = digit+ >start_token %set_media_port;
+        
+        media_nb_ports = digit+ >start_token %set_media_port;
+
+        # typically an RTP payload type for audio and video media
+        media_fmt = token >start_token %add_media_fmt;
+
+        media_description = (media_type ' ' media_port ('/' media_nb_ports)? ' ' media_proto (' ' media_fmt)+)
```

Comment:
Does SDP allows only a single space as delimiter?

Reply:
According to the [RFC 5234](https://tools.ietf.org/html/rfc5234#page-14) SP represents a single space. I added a definition in the parser.